### PR TITLE
common: add function ErrorWithContext

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Adds verbose output to error using runtime package information. 

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: common/common_test.go [hunk 1]
```diff
@@ -856,3 +856,36 @@ func TestErrorCollector(t *testing.T) {
...
+	runtimeCaller = runtime.Caller
+	runtimeFuncForPC = func(uintptr) *runtime.Func {
+		return nil
+	}
+
+	got = ErrorWithContext(err)
+	exp = []string{"common_test.go", "883", "internal test error"}
+	for _, e := range exp {
+		require.Contains(t, got.Error(), e)
+	}
```
Comment (gloriousCode) [inline common/common_test.go:887]: Please don't make this test fail due to the line numbers. Almost any changes to this file will cause this to fail. Just check that it's a number if you must do something with it.

**Review by gloriousCode — CHANGES_REQUESTED** [review]
I nice little helper for when developing things, but I wouldn't really want to see this around in main codebase

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Thanks for updating those tests 🌈

**Review by gloriousCode — APPROVED** [review]
ACK changes 🌈 
[Interesting](https://github.com/thrasher-corp/gocryptotrader/actions/runs/9951504826/job/27491117129?pr=1567#step:10:288) [races](https://github.com/thrasher-corp/gocryptotrader/actions/runs/9951504826/job/27491117516?pr=1567#step:10:2411)!

**Comment by shazbert** [comment]
Opted to not introduce code that is not integrated into library and never will be. Will open a PR to assign at the logger level if the the caller functionality is needed.

**Pull request closed without merging**